core/embed/prodtest/main.c

```diff
@@ -513,9 +603,443 @@ static void test_otp_write_device_variant(const char *args) {
+   *
+  if (OPTIGA_SUCCESS == ret) {
+    secret_erase();
+    secret_write_header();
+    secret_write(secret, SECRET_OPTIGA_KEY_OFFSET, SECRET_OPTIGA_KEY_LEN);
+  }
+
+  // Verify whether the secret was stored correctly in flash and OPTIGA.
+  memzero(secret, sizeof(secret));
+  if (secret_read(secret, SECRET_OPTIGA_KEY_OFFSET, SECRET_OPTIGA_KEY_LEN) !=
+      sectrue) {
+    vcp_println("ERROR Failed to read pairing secret.");
+    return false;
+  }
+  */
+
+  ret = optiga_sec_chan_handshake(secret, sizeof(secret));
+  memzero(secret, sizeof(secret));
+  if (OPTIGA_SUCCESS != ret) {
+    vcp_println("ERROR optiga_sec_chan_handshake error %d.", ret);
+    return false;
+  }
+
+  return true;
+}
+
+static void optiga_lock(void) {
+  if (!pair_optiga()) {
+    return;
+  }
+
+  // Delete trust anchor.
+  optiga_result ret =
+      optiga_set_data_object(0xe0e8, false, (const uint8_t *)"\0", 1);
+  if (OPTIGA_SUCCESS != ret) {
+    vcp_println("ERROR optiga_set_data error %d for 0xe0e8.", ret);
+    return;
+  }
+
+  // Set data object metadata.
+  static const optiga_metadata_item ACCESS_PAIRED = {
+      (const uint8_t *)"\x20\xE1\x40", 3};
+  static const optiga_metadata_item KEY_USE_SIGN = {(const uint8_t *)"\x10", 1};
+  static const optiga_metadata_item TYPE_PTFBIND = {(const uint8_t *)"\x22", 1};
+  optiga_metadata metadata = {0};
+
+  // Set metadata for device certificate.
+  memzero(&metadata, sizeof(metadata));
+  metadata.lcso = OPTIGA_LCS_OPERATIONAL;
+  metadata.change = OPTIGA_ACCESS_NEVER;
+  metadata.read = OPTIGA_ACCESS_ALWAYS;
+  metadata.execute = OPTIGA_ACCESS_ALWAYS;
+  if (!set_metadata(OID_CERT_DEV, &metadata)) {
+    return;
+  }
+
+  // Set metadata for FIDO attestation certificate.
+  memzero(&metadata, sizeof(metadata));
+  metadata.lcso = OPTIGA_LCS_OPERATIONAL;
+  metadata.change = OPTIGA_ACCESS_NEVER;
+  metadata.read = OPTIGA_ACCESS_ALWAYS;
+  metadata.execute = OPTIGA_ACCESS_ALWAYS;
+  if (!set_metadata(OID_CERT_FIDO, &metadata)) {
+    return;
+  }
+
+  // Set metadata for device private key.
+  memzero(&metadata, sizeof(metadata));
+  metadata.lcso = OPTIGA_LCS_OPERATIONAL;
+  metadata.change = OPTIGA_ACCESS_NEVER;
+  metadata.read = OPTIGA_ACCESS_NEVER;
+  metadata.execute = ACCESS_PAIRED;
+  metadata.key_usage = KEY_USE_SIGN;
+  if (!set_metadata(OID_KEY_DEV, &metadata)) {
+    return;
+  }
+
+  // Set metadata for FIDO attestation private key.
+  memzero(&metadata, sizeof(metadata));
+  metadata.lcso = OPTIGA_LCS_OPERATIONAL;
+  metadata.change = OPTIGA_ACCESS_NEVER;
+  metadata.read = OPTIGA_ACCESS_NEVER;
+  metadata.execute = ACCESS_PAIRED;
+  metadata.key_usage = KEY_USE_SIGN;
+  if (!set_metadata(OID_KEY_FIDO, &metadata)) {
+    return;
+  }
+
+  // Set metadata for pairing key.
+  memzero(&metadata, sizeof(metadata));
+  metadata.lcso = OPTIGA_LCS_OPERATIONAL;
+  metadata.change = OPTIGA_ACCESS_NEVER;
+  metadata.read = OPTIGA_ACCESS_NEVER;
+  metadata.execute = OPTIGA_ACCESS_ALWAYS;
```

Comment:
This slot is used only in the secure channel, it is not explicitly used. Is that true?

I was unable to find in the documentation whether execute permission is required for these purposes.

Reply:
It is used only in the secure channel. Execute permission is required.